server/api.go

```diff
@@ -314,29 +315,41 @@ func (a *API) autocompleteChannels(w http.ResponseWriter, r *http.Request) {
 }
 
 func (a *API) connect(w http.ResponseWriter, r *http.Request) {
+	connectURL := a.getConnectURL(w, r)
+	if connectURL != "" {
+		data, _ := json.Marshal(map[string]string{"connectUrl": connectURL})
+		_, _ = w.Write(data)
+	}
+}
+
+func (a *API) reconnect(w http.ResponseWriter, r *http.Request) {
+	connectURL := a.getConnectURL(w, r)
+	if connectURL != "" {
+		http.Redirect(w, r, connectURL, http.StatusSeeOther)
+	}
+}
```

Comment:
nit, 2/5

I wonder if `/connect?format=json` & `/connect` would be more appropriate than `/connect` & `/reconnect`?

Seeing `reconnect` in the invite URL `You're invited ... [Click here to connect your acccount](../reconnect)` may seem a bit off to some users.


On the other hand, do we even need to return the raw connect URL? Perhaps a single `/connect` that does the redirect would be enough? https://github.com/mattermost/mattermost-plugin-msteams/blob/21b2e41752dbd663981ea7015c8dee3d46778d63/webapp/src/index.tsx#L24-L26 
^ could turn into:
```tsx
onClick: () => window.open(Client.getConnectRedirectUrl(), '_blank');
```